File: SwordDemoLibrary/codelibrary/src/main/java/com/sword/SwordApplication.java
package com.sword;

import android.app.Application;
import android.content.Context;

public class SwordApplication extends Application {
  private static Context applicationContext;
  
  @Override
  public void onCreate() {
    super.onCreate();
    
    applicationContext = getApplicationContext();
  }
  
  public static Context getGlobalContext() {
    return applicationContext;
  }
}

File: SwordDemoLibrary/app/src/main/java/sword/SwordApplication.kt
package sword

import android.annotation.SuppressLint
import android.app.Application
import android.content.Context
import leakcanary.LeakCanary

class SwordApplication : Application() {
  override fun onCreate() {
    super.onCreate()
    globalContext = applicationContext
  }

  companion object {
    @SuppressLint("StaticFieldLeak")
    @JvmStatic
    var globalContext: Context? = null
      private set
  }
}